test/demo/raiseIntent.inttest.ts

```diff
@@ -371,85 +368,393 @@ only the first listener is triggered exactly once with the correct context, and
+            await delay(500);
+            reject(e);
         });
+        await closeResolver();
+    });
+}
 
-        // TODO: find a way to dynamically set the app directory for these tests
-        describe('With exactly one app registered to accept the raised intent', () => {
-            describe('And no running ad-hoc apps with listeners registered for the raised intent', () => {
-                describe('With the registered app running', () => {
-                    test.todo('When calling raiseIntent from another app the listener is triggered exactly once with the correct context');
+async function test_raiseIntentExpectResolverSelectApp(title: string, getParams: () =>
+    [Intent, AppIdentity, fdc3Remote.RemoteIntentListener?]) {
+    test(title, async () => {
+        const [intent, app, listener] = getParams();
+        const [raiseIntentPromise] = await raiseIntentAndExpectResolverToShow(intent);
+        await selectResolverApp(app.name);
+        await raiseIntentPromise; // Now the intent resolves
 
-                    test.todo('But the app does not have the listener registered on the model');
-                });
+        // If no intent listener provided, try to fetch it "live"
+        const remoteListener = listener || await fdc3Remote.getRemoteIntentListener(app, intent.type);
 
-                describe('With the registered app not running', () => {
-                    test.todo('The targeted app opens and its listener is triggered exactly once with the correct context');
+        const receivedContexts = await remoteListener.getReceivedContexts();
+        expect(receivedContexts).toEqual([intent.context]);
+    });
+}
+
+/**
+ * Closes the resolver by remotely clicking the Cancel button in it
+ */
+async function closeResolver() {
+    const cancelClicked = await fdc3Remote.clickHTMLElement(resolverWindowIdentity, '#btn-cancel');
+    if (!cancelClicked) {
+        throw new Error('Error clicking cancel button on resolver. Make sure it has id="btn-cancel".');
+    }
+    await delay(100); // Give the UI some time to process the click and close the window
+
+    const isResolverShowing = await fin.Window.wrapSync(resolverWindowIdentity).isShowing();
+    expect(isResolverShowing).toBe(false);
+}
+
+/**
+ * Selects an app on the resolver by remotely clicking on its button and subsequently on the 'open' button
+ * @param appName name of app to open
+ */
+async function selectResolverApp(appName: string) {
+    const appClicked = await fdc3Remote.clickHTMLElement(resolverWindowIdentity, `#${appName}`);
+    if (!appClicked) {
+        throw new Error(`App with name '${appName}' not found in resolver`);
+    }
+    await delay(100);
+    const openClicked = await fdc3Remote.clickHTMLElement(resolverWindowIdentity, '#btn-open');
+    if (!openClicked) {
+        throw new Error('Error clicking open button on resolver. Make sure it has id="btn-open".');
+    }
+    await delay(100);
```

Comment:
If we need this delay after every click, could it go inside clickHTMLElement

Reply:
The delay after each click could be specific to the action in particular e.g. the click to select the app should need less time than the click on the `Open` button, after which we wait for the resolver window to close. I didn't put much thought on the values here, they just seemed to work :)
Perhaps it would still be a good idea to add a small delay on `clickHTMLElement` itself to at least give React some time to propagate UI changes - that way we could omit explicit delays after simple clicks (like the first one in this function), and let consumers add an additional delay if required (like the second one)